libraries/Microsoft.Bot.Expressions/Memory/SimpleObjectMemory.cs

```diff
@@ -36,9 +36,9 @@ public static IMemory Wrap(object obj)
         public bool TryGetValue(string path, out object value)
         {
             value = null;
-            if (memory == null)
+            if (memory == null || path.Count() < 1 || (path[0] != '[' && !char.IsLetter(path[0])))
```

Comment:
>Count [](start = 39, length = 5)

path.Length == 0?

Reply:
what is this check for?

---
In reply to: [363905784](https://github.com/microsoft/botbuilder-dotnet/pull/3226#discussion_r363905784) [](ancestors = 363905784)

Reply:
Local paths cannot start with anything other than a letter.  ==0 would be fine and I can change that.

---
In reply to: [363906193](https://github.com/microsoft/botbuilder-dotnet/pull/3226#discussion_r363906193) [](ancestors = 363906193,363905784)